scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1391,9 +1391,12 @@ private void processHandshakeRecord(final Record record, final Connection connec
 				// We do not support re-negotiation as recommended in :
 				// https://tools.ietf.org/html/rfc7925#section-17
 				if (record.getEpoch() > 0) {
```

Comment:
I asked my self if this `if (record.getEpoch() > 0) {` is really needed as some line above we check that this is not a new CLIENT_HELLO ?

Reply:
> as some line above we check that this is not a new CLIENT_HELLO ?

do you mean 

```
public boolean isNewClientHello() {
	if (0 < epoch || type != ContentType.HANDSHAKE || 0 == fragmentBytes.length) {
		return false;
	}
	HandshakeType handshakeType = HandshakeType.getTypeByCode(fragmentBytes[0]);
	return handshakeType == HandshakeType.CLIENT_HELLO;
}
```

that is only for epoch 0 CLIENT_HELLOs. So my feeling is, this check seems to be not too bad.

Reply:
OK, I got it. Yes, it look like it' obsolete in the meantime.